cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/ResourceDistributionGoal.java

```diff
@@ -574,7 +590,7 @@ private boolean rebalanceBySwappingLoadIn(Broker broker,
       candidateBrokerPQ.add(candidateBroker);
     }
 
-    while (!candidateBrokerPQ.isEmpty()) {
+    while (!candidateBrokerPQ.isEmpty() && remainingPerBrokerSwapTimeMs(swapStartTimeMs) > 0) {
```

Comment:
Could we move `remainingPerBrokerSwapTimeMs(swapStartTimeMs)` inside the loop and make it in an `if` statement. and if the condition holds, we do the same `LOG.debug("Swap load in timeout for broker {}.", broker.id());`. I feel it would help in we want to debug the optimization logic in the future in that under all scenario a message will get printed if we have optimization failed due to timeout.